app/services/npq/users/find_or_create_by.rb

```diff
@@ -0,0 +1,119 @@
+# frozen_string_literal: true
+
+module NPQ
```

Comment:
Largely it's down to the errors not being actually about a specific user. The conflict errors aren't about one user or another, but both and the request itself. And some don't even have a user to refer to, for example the missing email error that can be returned.

By using this object being returned that responds to both user and error, returning a value or nil for each allows to have a single interface for all types of errors abstracted away from the specific cause. You get an object that contains a user or error like objects.

Using an OpenStruct over a Hash is really just a quirk of how I implemented it, they're functionally identical except `response.user.present?` feels a little bit nicer than `response[:user].present?`.

Partially also I was aiming to preserve the single call method on the class as we have previously discussed, rather than having the instantiated object itself response to `user` and `errors`.